compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp

```diff
@@ -376,6 +376,95 @@ INTERCEPTOR(void, setbuffer, FILE *stream, char *buf, int size) {
 #define RTSAN_MAYBE_INTERCEPT_SETBUFFER
 #endif
 
+#if SANITIZER_INTERCEPT_FSEEK
```

Comment:
This is more of a question - I don't have the answer. Should all of these be inside this ifdef? Should some of them be outside of it? 

For instance, on my mac, I have access to many of these:

```
FSEEK(3)                   Library Functions Manual                   FSEEK(3)

NAME
     fgetpos, fseek, fseeko, fsetpos, ftell, ftello, rewind – reposition a
     stream
```

And it seems like these only have support for BSD flavors:
```
#define SANITIZER_INTERCEPT_FSEEK (SI_NETBSD || SI_FREEBSD)

```

Even normally flavored linux has many of these https://www.man7.org/linux/man-pages/man3/fseek.3.html

Maybe this macro needs updating?

Reply:
oh yes weird, I m pretty sure even fuchsia has it too, but setting as SI_POSIX should do at first.